### PR TITLE
Issue 2 relative vendordir

### LINKED ISSUE #2: luya symlink is created as absolute path
When creating a luya project, the `luya` symlink in the project directory is created as an absolute path, e.g.:

```
$ ls -la |grep luya
lrwxrwxrwx  1 cebehosting cebehosting    61 Nov  7 09:40 luya -> /home/cebe/dev/cebe.cc/luya/vendor/luyadev/luya-core/bin/luya
```

when deploying this repo to a production environment, the link is not correct anymore. ideally it should be created relative to the current path:

```
luya -> vendor/luyadev/luya-core/bin/luya
```

the code for this is here:

https://github.com/luyadev/luya-composer/blob/7cd5ceb315c151a65d5f10e61259d332fe4394ec/src/Plugin.php#L57-L59

### PR DESCRIPTION
**Pull request by boehsermoe**
closes #2

### CODE REVIEW

**Comment by nadar** [comment]
I am not sure if `@vendor` is a good solution, cause if we are once out of application context the path wont work. Why not using relativ paths for both?

**Comment by nadar** [comment]
Besides: I need time to test, cause when i was working on this it was very hard to have good unit tests or even test the plugin at all. Composer plugins are really hard to test (at least as far as i can see). So maybe they changed this or i need to invest more time in testing.

**Comment by boehsermoe** [comment]
Ok better relative path too. I will take look to the unit tests and will expend them if necessary.

**Comment by boehsermoe** [comment]
I refactored and advanced the unit test. I know it is much stuff in `TestCase::setUp` but I took it over from the composer.

**Review by nadar — APPROVED** [review]
Looks good to me at the first sight. I will make some tests and merge afterwards.

**Comment by nadar** [comment]
(@boehsermoe i am really sorry i had no time to take care on this yet, will do it next week!)

**Comment by boehsermoe** [comment]
@nadar some weeks ago now 😉

**Comment by nadar** [comment]
Still in my todo :-) thanks for mention

**Comment by nadar** [comment]
I will setup travis, but i strongly assume you did run the test successfully and correct of course. In generell, i really had problems with unit testing for composer plugin, this is just a pain (or maybe i did not understand correctly how to test "real word scenarios") - therefore i also created this "how to test" guide section - https://github.com/luyadev/luya-composer#local-testing-for-composer-plugin-development have you used this?

File: src/Installer.php [hunk 1]
```diff
@@ -137,4 +138,15 @@ protected function writeInstaller(array $data)
             @opcache_invalidate($file, true);
         }
     }
+    
+    public function getRelativeVendorDir()
```
Comment (nadar) [inline src/Installer.php:148]: @boehsermoe please add php docs and since tag.

File: src/Plugin.php [hunk 2]
```diff
@@ -24,21 +24,28 @@ class Plugin implements PluginInterface, EventSubscriberInterface
...
     private $_packageInstalls = [];
     
-    private $_vendorDir = null;
+    private $_relativeVendorDir = null;
     
     protected $io;
     
     protected $composer;
     
+    public $linkPath = 'luya';
```
Comment (nadar) [inline src/Plugin.php:38]: please add phpdoc and since tags

File: src/Plugin.php [hunk 3]
```diff
@@ -24,21 +24,28 @@ class Plugin implements PluginInterface, EventSubscriberInterface
...
+    
+    public function getRelativeVendorDir(Composer $composer)
```
Comment (nadar) [inline src/Plugin.php:46]: Same here: php doc and since tag

File: tests/LuyaComposerPluginTest.php [hunk 4]
```diff
@@ -13,46 +12,48 @@ class LuyaComposerPluginTest extends TestCase
...
     {
         $this->plugin->activate($this->composer, $this->io);
         
-        $this->assertCount(3, $this->plugin->getSubscribedEvents());
+        $this->assertCount(4, $this->plugin->getSubscribedEvents());
         
         $packageEvent = $this->getMockBuilder(PackageEvent::class)->disableOriginalConstructor()->getMock();
-        
+    
+        // @todo: own test
```
Comment (nadar) [inline tests/LuyaComposerPluginTest.php]: @boehsermoe what does that mean? So i can not merge unless there is something to do :-)

**Review by nadar — CHANGES_REQUESTED** [review]
I was running the tests locally, they are passing. Travis errors, not sure why. Please add the php docs and missing since tags. Also there is a todo  - maybe we should do the to-do? :-)

Otherwise looks good.

**Review by nadar — APPROVED** [review]

**Pull request merged**